server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationMailet.java

```diff
@@ -77,20 +82,36 @@ public void service(Mail mail) {
     private boolean shouldSendNotification(Vacation vacation, Mail processedMail, MailAddress recipient, ZonedDateTime processingDate) {
         try {
             return vacation.isActiveAtDate(processingDate)
-                && ! automaticallySentMailDetector.isAutomaticallySent(processedMail);
+                && ! automaticallySentMailDetector.isAutomaticallySent(processedMail)
+                && hasNotSentNotificationsYet(processedMail, recipient);
         } catch (MessagingException e) {
             LOGGER.warn("Failed detect automatic response in a mail from {} to {}", processedMail.getSender(), recipient, e);
             return false;
         }
     }
 
+    private boolean hasNotSentNotificationsYet(Mail processedMail, MailAddress recipient) {
+        CompletableFuture<Boolean> hasAlreadyBeenSent = notificationRegistry.isRegistered(
+            AccountId.fromString(recipient.toString()),
+            RecipientId.fromMailAddress(processedMail.getSender()));
+        try {
+            return !hasAlreadyBeenSent.join();
+        } catch (Throwable t) {
```

Comment:
No problem with you logic. That was still a pure Java question, between Throwable and Exception. Let's discuss about it on another channel.